compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/ModuleLevelVariableDeclarationAnalysisTask.java

```diff
@@ -76,6 +78,13 @@ public void perform(SyntaxNodeAnalysisContext context) {
         Schema schema = generateSchema(context, interfaceFinder, graphqlServiceObjectNode, description);
         DocumentId documentId = context.documentId();
         addToModifierContextMap(documentId, moduleVariableDeclarationNode, schema);
+
+        Project project = context.currentPackage().project();
+        VariableSymbol serviceVariableSymbol =
+                (VariableSymbol) context.semanticModel().symbol(moduleVariableDeclarationNode).get();
+        SDLFileGenerator sdlFileGenerator =
+                new SDLFileGenerator(schema, serviceVariableSymbol.hashCode(), project);
+        sdlFileGenerator.generate();
```

Comment:
Let's move the schema file generation logic to code modifier so that we won't duplicate the code and also to keep track of the file names with the service.